src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java

```diff
@@ -127,10 +128,13 @@ public SecurityContextRepository contextRepository(ReactiveAuthenticationManager
 
         private Mono<SecurityContext> checkCentralRegistry(String token) {
             return centralRegistryTokenVerifier.verify(token)
-                    .map(caller -> new UsernamePasswordAuthenticationToken(
-                            caller,
-                            token,
-                            new ArrayList<SimpleGrantedAuthority>()))
+                    .map(serviceCaller -> {
+                        var authority = serviceCaller.getRoles()
+                                .stream()
+                                .map( role -> new SimpleGrantedAuthority("ROLE_" + role.name().toUpperCase()))
+                                .collect(toList());
+                        return new UsernamePasswordAuthenticationToken(serviceCaller, token, authority);
+                    })
```

Comment:
As authority holds a list.. u can rename it to authorities:)

Reply:
changed it to authorities